test/common_test/joint_features.cc

```diff
@@ -1138,7 +1138,7 @@ TYPED_TEST(JointFeaturesAttachDetachTest, JointAttachDetach)
     }
     frameDataModel2Body = model2Body->FrameDataRelativeToWorld();
 
-    EXPECT_NEAR(0.0, frameDataModel2Body.linearVelocity.z(), 1e-3);
+    EXPECT_NEAR(0.0, frameDataModel2Body.linearVelocity.z(), 2e-3);
```

Comment:
nit: this check seems a bit flaky, if we update erp2 in future, we'll have to tune this threshold again. Given that this test has nothing to do with collisions, I'd recommend checking for a combination of position and velocity thresholds instead inside the loop above, i.e.

```
while (true)
{
  world->Step(output, state, input);
  frameDataModel2Body = model2Body->FrameDataRelativeToWorld();
  // Expected Z height of model2 is 0.75 when both boxes are stacked on top of each other since each is 0.5 high.
  if (fabs(frameDataModel2Body.pose.z() - 0.75) < 2e-2 && fabs(frameDataModel2Body.linearVelocity.z()) < 1e-3)
  {
    break;
  }
}
```
Ideally, the position threshold of 1e-2 above can be set based on the configured margin for both objects, but it should be ok to use a large value.

Reply:
thanks for the suggestions! I updated to use while loop with max step count to make sure the loop eventually exits. [8ea3e67](https://github.com/gazebosim/gz-physics/pull/600/commits/8ea3e67e90177a28a5e6ed31da02dbec7bb93415)